File: app/src/main/java/net/lyxnx/carcheck/InfoActivity.java
package net.lyxnx.carcheck;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.TableLayout;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public abstract class InfoActivity extends AppCompatActivity {

    @Override
    public void onBackPressed() {
        finish();
    }

    protected void addToTable(TableLayout table, String header, String value) {
        View v = LayoutInflater.from(this).inflate(R.layout.row_layout, null);

        setText(v.findViewById(R.id.rowHeader), header);
        setText(v.findViewById(R.id.rowValue), value);

        table.addView(v);
    }

    protected void setText(TextView textView, String text) {
        textView.setText(text);
    }
}

File: app/src/main/java/net/lyxnx/carcheck/InfoActivity.kt
package net.lyxnx.carcheck

import android.widget.TableLayout
import androidx.appcompat.app.AppCompatActivity
import net.lyxnx.carcheck.widgets.InfoTableRow

open class InfoActivity : AppCompatActivity() {

    override fun onBackPressed() {
        finish()
    }

    fun TableLayout.add(header: String, value: String) {
        val row = InfoTableRow(context)

        row.setHeader(header)
        row.setValue(value)

        addView(row)
    }
}